src/compile/output/mod.rs

```diff
@@ -32,6 +33,8 @@ pub struct ProjectCompileOutput<T: ArtifactOutput = ConfigurableArtifacts> {
     pub(crate) cached_artifacts: Artifacts<T::Artifact>,
     /// errors that should be omitted
     pub(crate) ignored_error_codes: Vec<u64>,
+    /// paths that should be omitted
+    pub(crate) ignored_file_paths: ProjectPaths,
```

Comment:
I think we either want a Vec<PathBuf> or an Option here

because now it's not obvious to distinguish between "include library errors" or exclude them

the Vec<PathBuf> solution is likely more flexible 

if configured, all warnings that originate in a path will be skipped